lib/net/pool.js

```diff
@@ -3390,6 +3392,10 @@ class Pool extends EventEmitter {
       if (this.options.brontideOnly && !addr.hasKey())
         continue;
 
+      // Don't connect to outbound peers in the same group.
+      if (this.connectedGroups.has(addr.getGroup()))
+        continue;
```

Comment:
should we allow this with some low probability? or is it always bad

Reply:
hm.. Maybe last 2 tries ? pc80 ?

Reply:
looks like Bitcoin Core really doesnt allow this at all:

https://github.com/bitcoin/bitcoin/blob/master/src/net.cpp#L2114-L2117

So I defer to you, just not sure given the limited size of the handshake network if this is actually beneficial.